### PR TITLE
adding separate resource section for aws_s3_bucket_acl & server_side encryption

### PR DESCRIPTION
**Pull request by prem-fissionhq**
https://dimagi-dev.atlassian.net/browse/SAAS-13313
##### ENVIRONMENTS AFFECTED

All

Adding comment lines only for this review and once approved I will remove them if suggested.

--
When these resources were added in .tf file we don't have to apply them since they are already, instead we have to import them using terraform import commands. 
Whereas, for acl it is asking us to apply the change even after importing. And I see that it is safe to apply,

### CODE REVIEW

File: src/commcare_cloud/terraform/modules/logshipping/main.tf [hunk 1]
```diff
@@ -8,16 +8,28 @@ locals {
...
 resource "aws_s3_bucket" "log_bucket" {
   bucket = local.log_bucket_name
-  acl = "private"
+  #https://github.com/hashicorp/terraform-provider-aws/issues/23888
+  lifecycle {
+    ignore_changes = [
+    acl,
+    server_side_encryption_configuration
+  ]
+}
```
Comment (dannyroberts) [inline src/commcare_cloud/terraform/modules/logshipping/main.tf:17]: I think these lines should be indented
```suggestion
      acl,
      server_side_encryption_configuration
    ]
  }
```

File: src/commcare_cloud/terraform/modules/logshipping/main.tf [hunk 2]
```diff
@@ -8,16 +8,28 @@ locals {
...
+}
+#https://registry.terraform.io/providers/hashicorp/aws/3.75.1/docs/resources/s3_bucket_server_side_encryption_configuration#usage-notes
```
Comment (dannyroberts) [inline src/commcare_cloud/terraform/modules/logshipping/main.tf:19]: Nit
```suggestion

# https://registry.terraform.io/providers/hashicorp/aws/3.75.1/docs/resources/s3_bucket_server_side_encryption_configuration#usage-notes
```

File: src/commcare_cloud/terraform/modules/logshipping/main.tf [hunk 3]
```diff
@@ -8,16 +8,28 @@ locals {
...
-      apply_server_side_encryption_by_default {
-        sse_algorithm = "AES256"
-      }
+  rule {
+    apply_server_side_encryption_by_default {
+      sse_algorithm = "AES256"
     }
   }
 }
+resource "aws_s3_bucket_acl" "log_bucket" {
```
Comment (dannyroberts) [inline src/commcare_cloud/terraform/modules/logshipping/main.tf:29]: Nit
```suggestion

resource "aws_s3_bucket_acl" "log_bucket" {
```

**Review by dannyroberts — COMMENTED** [review]
Looks good! I made some minor formatting suggestions.

Is the idea that this should work with both aws provider versions 3.x and 4.x? And you'll update to 4.x in a following commit?

File: src/commcare_cloud/terraform/modules/logshipping/main.tf [hunk 4]
```diff
@@ -8,17 +8,32 @@ locals {
...
-      apply_server_side_encryption_by_default {
-        sse_algorithm = "AES256"
-      }
+  # https://github.com/hashicorp/terraform-provider-aws/issues/23888
+  lifecycle {
+      ignore_changes = [
+      acl,
+      server_side_encryption_configuration
+  ]
+}
```
Comment (dannyroberts) [inline src/commcare_cloud/terraform/modules/logshipping/main.tf:18]: I might have given you the wrong suggestion b/c it still looks off. Try two
```suggestion
  lifecycle {
    ignore_changes = [
      acl,
      server_side_encryption_configuration
    ]
  }
```

**Review by dannyroberts — APPROVED** [review]

**Comment by prem-fissionhq** [comment]
`Is the idea that this should work with both aws provider versions 3.x and 4.x? And you'll update to 4.x in a following PR?` 
@dannyroberts update to use 4.x and it is working fine.

**Review by dannyroberts — APPROVED** [review]
let me or @shyamkumarlchauhan know and we can merge

**Comment by prem-fissionhq** [comment]
> let me or @shyamkumarlchauhan know and we can merge

Sure Danny.

**Pull request merged**